qt/scientific_interfaces/Indirect/test/ISISEnergyTransferModelTest.h

```diff
@@ -0,0 +1,605 @@
+// Mantid Repository : https://github.com/mantidproject/mantid
+//
+// Copyright &copy; 2023 ISIS Rutherford Appleton Laboratory UKRI,
+//   NScD Oak Ridge National Laboratory, European Spallation Source,
+//   Institut Laue - Langevin & CSNS, Institute of High Energy Physics, CAS
+// SPDX - License - Identifier: GPL - 3.0 +
+#pragma once
+
+#include "IndirectDataValidationHelper.h"
+#include "MantidAPI/Algorithm.h"
+#include "MantidAPI/AlgorithmManager.h"
+#include "MantidAPI/AnalysisDataService.h"
+#include "MantidAPI/ITableWorkspace.h"
+#include "MantidAPI/TableRow.h"
+#include "MantidFrameworkTestHelpers/WorkspaceCreationHelper.h"
+#include "MantidQtWidgets/Common/BatchAlgorithmRunner.h"
+#include "Reduction/ISISEnergyTransferData.h"
+#include "Reduction/ISISEnergyTransferModel.h"
+
+#include <chrono>
+#include <cxxtest/TestSuite.h>
+#include <gmock/gmock.h>
+#include <thread>
+
+using namespace Mantid::API;
+using namespace Mantid::Kernel;
+using namespace MantidQt::CustomInterfaces;
+
+class ISISIndirectEnergyTransferWrapper : public Algorithm {
+public:
+  const std::string name() const override { return "ISISIndirectEnergyTransferWrapper"; };
+  int version() const override { return 1; };
+  const std::string summary() const override { return "ISISIndirectEnergyTransferWrapper Mock algorithm"; };
```

Comment:
why are we not using mock?

Reply:
hmm, is it possible to mock a class that inherits another class? I followed this approach as it is consistent with how other indirect data analysis tests are written

Reply:
I see no reason why it wouldn't be possible. Lets leave it for now, since its what the others do